### PR TITLE
List configurable parameters for rules

### LINKED ISSUE #207: List rules with available configurable parameters
It would be very useful to have some kind of cli option to list all configurables for all rules and maybe it could also support glob patterns so you can e.g. list only documentation configurables: `--list-configurables *doc*` 

_Originally posted by @mnojek in https://github.com/MarketSquare/robotframework-robocop/issues/201#issuecomment-694872611_

### PR DESCRIPTION
**Pull request by bhirsz**
Add --list-configurables option that prints out all rules together with configurable parameters
Add support for --list [PATTERN] and --list-configurables [PATTERN] that filter outs printed list by given glob pattern

Closes #207

### CODE REVIEW

**Comment by bhirsz** [comment]
I need also to update README and docs

**Review by mnojek — DISMISSED** [review]

File: docs/user_guide.rst [hunk 1]
```diff
@@ -16,6 +16,39 @@ Including or excluding rules
...
+
+    robocop --exclude 02* --list
+    Rule - 0201 [W]: missing-doc-keyword: Missing documentation in keyword (disabled)
+    Rule - 0202 [W]: missing-doc-testcase: Missing documentation in test case (disabled)
+    Rule - 0203 [W]: missing-doc-suite: Missing documentation in suite (enabled)
+    (...)
+
+Rules list can be filtered out by glob pattern::
+
+    robocop --list setting-name*
```
Comment (mnojek) [inline docs/user_guide.rst:39]: This example does not correspond to the output. It should be:
```suggestion
    robocop --list tag*
```

File: docs/user_guide.rst [hunk 2]
```diff
@@ -16,6 +16,39 @@ Including or excluding rules
...
+    Rule - 0601 [W]: tag-with-space: Tags should not contain spaces (enabled)
+    Rule - 0602 [I]: tag-with-or-and: Tag with reserved word OR/AND. Hint: make sure to include this tag using lowercase name to avoid issues (enabled)
+    Rule - 0603 [W]: tag-with-reserved: Tag prefixed with reserved word `robot:`. Only allowed tag with this prefix is robot:no-dry-run (enabled)
+    Rule - 0606 [I]: tag-already-set-in-force-tags: This tag is already set by Force Tags in suite settings (enabled)
+
+ Use --list-configurables argument to list rules together with available configurable parameters. Optional pattern argument is also supported::
```
Comment (mnojek) [inline docs/user_guide.rst:45]: There is a trailing line at the beginning of this line.

Comment (mnojek) [reply docs/user_guide.rst:45]: Also put option name in double ticks like this: ``--list-configurables``

File: docs/user_guide.rst [hunk 3]
```diff
@@ -16,6 +16,39 @@ Including or excluding rules
 Rules can be included or excluded from command line. It is also possible to disable rule(s) from Robot Framework
 source code. More in :ref:`including-rules`.
 
+Listing available rules
+-----------------------
+To get list of available rules (with enabled/disabled status) use --list option::
```
Comment (mnojek) [inline docs/user_guide.rst:21]: Format it: ``--list``

File: docs/user_guide.rst [hunk 4]
```diff
@@ -16,6 +16,39 @@ Including or excluding rules
...
+    robocop --list setting-name*
+    Rule - 0601 [W]: tag-with-space: Tags should not contain spaces (enabled)
+    Rule - 0602 [I]: tag-with-or-and: Tag with reserved word OR/AND. Hint: make sure to include this tag using lowercase name to avoid issues (enabled)
+    Rule - 0603 [W]: tag-with-reserved: Tag prefixed with reserved word `robot:`. Only allowed tag with this prefix is robot:no-dry-run (enabled)
+    Rule - 0606 [I]: tag-already-set-in-force-tags: This tag is already set by Force Tags in suite settings (enabled)
+
+ Use --list-configurables argument to list rules together with available configurable parameters. Optional pattern argument is also supported::
+
+    robocop --list-configurables empty-lines-between-section
+    Rule - 1003 [W]: empty-lines-between-sections: Invalid number of empty lines between sections (%d/%d) (enabled). Available configurable(s) for this rule:
```
Comment (mnojek) [inline docs/user_guide.rst:48]: I see now how it looks in the output and maybe we can move the last sentence ("Available configurable(s) for this rule:") to the next line?

File: docs/user_guide.rst (hunk 2 above)
Comment (bhirsz) [reply docs/user_guide.rst:45]: In .rst we need to add trailing line after code examples:

```
::

    this is code block

And this is next line
```

File: docs/user_guide.rst (hunk 4 above)
Comment (bhirsz) [reply docs/user_guide.rst:48]:
```
Rule - 1005 [W]: empty-lines-between-keywords: Invalid number of empty lines between keywords (%d/%d) (enabled)
Available configurable(s) for this rule:
severity
empty_lines
Rule - 1006 [W]: mixed-tabs-and-spaces: Inconsistent use of tabs and spaces in file (enabled)
Available configurable(s) for this rule:
severity
```

? Of course type of the configurable would be handy too but I will leave tinkering for later.

Comment (bhirsz) [reply docs/user_guide.rst:48]: Or with indent:
```
Rule - 1005 [W]: empty-lines-between-keywords: Invalid number of empty lines between keywords (%d/%d) (enabled)
    Available configurable(s) for this rule:
        severity
        empty_lines
Rule - 1006 [W]: mixed-tabs-and-spaces: Inconsistent use of tabs and spaces in file (enabled)
    Available configurable(s) for this rule:
        severity
```

Comment (mnojek) [reply docs/user_guide.rst:48]: Indent looks much better :)

**Review by mnojek — APPROVED** [review]

**Pull request merged**